### PR TITLE
New class and changes

### PR DESCRIPTION
**Pull request by knil-maloon**
- Added MessageHandler class to manage message display.
- Removed page refresh and unused SESSIONS.

### CODE REVIEW

File: REDCapInstrumentAndVariableSQLRenamer.php [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <?php
 
 namespace VUMC\REDCapInstrumentAndVariableSQLRenamer;
+include_once(__DIR__ . "/classes/MessagedHandler.php");
```
Comment (paulj3) [inline REDCapInstrumentAndVariableSQLRenamer.php]: It looks like you have a type in this class name / filename. It should be "MessageHandler", correct?

File: classes/MessagedHandler.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public function setPrintVariable($printVariable): void
+    {
+        $this->printVariable = $printVariable;
+    }
+
+    public function setPrintInstrument($printInstrument): void
+    {
+        $this->printInstrument = $printInstrument;
+    }
+}
```
Comment (paulj3) [inline classes/MessagedHandler.php]: You'll want to add an EOF line here

File: classes/MessagedHandler.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    public $printVariable = '';
+
+    public $printInstrument = '';
+
+    public function __construct(REDCapInstrumentAndVariableSQLRenamer $module)
+    {
+        $this->module = $module;
+    }
+
+    public function messageType($messageType): void
```
Comment (paulj3) [inline classes/MessagedHandler.php]:
```suggestion
    public function setMessageType($messageType): void
```

File: classes/MessagedHandler.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        $this->messageType = $messageType;
+    }
+
+    public function addMessage($message): void
```
Comment (paulj3) [inline classes/MessagedHandler.php]:
```suggestion
    public function setMessage($message): void
```

You could use "add" here, but when I see the word "add" on a class like this, I expect a 1 to many situation, where I'm adding one to an array.

File: saveData.php [hunk 5]
```diff
@@ -94,21 +95,22 @@
...
-    $_SESSION['message_type'] = "warning_message";
+    $module->getMessageHandler()->addMessage($message);
+    $module->getMessageHandler()->messageType('danger');
     throw $e;
 }
 
-echo json_encode(array(
-                     'status' => "success"
-                 ));
+echo json_encode($module->getMessageHandler());
```
Comment (paulj3) [inline saveData.php]: I find this line a little confusing. It looks like json_encoding a class.  What happens if new properties and methods are added to this class, and we don't want those to be encoded? Or if those have sensitive information? I would add a new message to your message handler like `getMessageAttributes` that returns an explicitly defined array of what the front end expects.

**Review by paulj3 — COMMENTED** [review]
This is an amazing PR! I see you reviewed Karen's work and tried some things out. I really like your initiative and your implementation, I just have a few notes I've left as comments. Let me know when you're ready for me again

File: REDCapInstrumentAndVariableSQLRenamer.php (hunk 1 above)
Comment (knil-maloon) [reply REDCapInstrumentAndVariableSQLRenamer.php]: Deleted this. It is not necessary but good catch!

File: classes/MessagedHandler.php (hunk 2 above)
Comment (knil-maloon) [reply classes/MessagedHandler.php]: Added!

File: classes/MessagedHandler.php (hunk 3 above)
Comment (knil-maloon) [reply classes/MessagedHandler.php]: Name changed

File: classes/MessagedHandler.php (hunk 4 above)
Comment (knil-maloon) [reply classes/MessagedHandler.php]: Name changed

File: saveData.php (hunk 5 above)
Comment (knil-maloon) [reply saveData.php]: Created the new class. I like this approach.

**Comment by knil-maloon** [comment]
Thank you! I had some time in between testing with Judy so I really wanted to try this out. I'm glad you tagged me to Karen's PR as I thought it was useful!

File: classes/MessagedHandler.php [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    public function setMessageType($messageType): void
+    {
+        $this->messageType = $messageType;
+    }
+
+    public function setMessage($message): void
+    {
+        $this->message = $message;
+    }
```
Comment (paulj3) [inline classes/MessagedHandler.php]: 👍

File: classes/MessagedHandler.php [hunk 7]
```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace VUMC\REDCapInstrumentAndVariableSQLRenamer;
+
+use REDCap;
+
+class MessagedHandler
```
Comment (paulj3) [inline classes/MessagedHandler.php]: I think you still need to rename this class and file 'MessageHandler" and not "Message**d**Handler"

**Review by paulj3 — APPROVED** [review]

**Pull request merged**